### PR TITLE
[SCB-562] fix NPE when had empty interface (not have any method)

### PR DESCRIPTION
**Pull request by zhengyangyong**
Signed-off-by: zhengyangyong <yangyong.zheng@huawei.com>

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---
do not init operations if there are no path (empty interface) in swagger(prevent NPE)

### CODE REVIEW

File: core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java [hunk 1]
```diff
@@ -83,40 +83,42 @@ public String getPackageName() {
...
 
   private void initOperations() {
-    for (Entry<String, Path> entry : swagger.getPaths().entrySet()) {
-      String strPath = entry.getKey();
-      Path path = entry.getValue();
-      for (Entry<HttpMethod, Operation> operationEntry : path.getOperationMap().entrySet()) {
-        Operation operation = operationEntry.getValue();
-        if (operation.getOperationId() == null) {
-          throw ExceptionUtils.operationIdInvalid(getSchemaId(), strPath);
+    if (swagger.getPaths() != null) {
```
Comment (WillemJiang) [inline core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java]: I think we to write a logger for it, so use can know what's wrong here.

Comment (zhengyangyong) [reply core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java]: I think no necessary, user want fill this empty interface later, so it's not a `wrong scenario`

Comment (wujimin) [inline core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java]: so big block
just return when null

Comment (zhengyangyong) [reply core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java]: good idea, done

Comment (zhengyangyong) [reply core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java]: emmm, ok , add an error output in order to notice user like
`swagger definition for org.apache.servicecomb.samples.uselog4j2.HelloImpl with path /hello is an empty interface, please delete it or fill with one method!`

File: core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java [hunk 2]
```diff
@@ -84,6 +84,8 @@ public String getPackageName() {
 
   private void initOperations() {
```
Comment (WillemJiang) [inline core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java]: It just does the check, maybe we can name it as checkOperations.

File: core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java [hunk 3]
```diff
@@ -83,6 +83,12 @@ public String getPackageName() {
   }
 
   private void initOperations() {
+    if (swagger.getPaths() == null) {
+      LOGGER.error(swagger.getInfo().getTitle() + " with path " + swagger.getBasePath()
```
[the remark this answers is not in the document]
Comment (zhengyangyong) [reply core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java]: Had considered as warn before, but may mark as error to let user know it is not an appropriate practice and fix as quick as possible ?

[the remark this answers is not in the document]
Comment (liubao68) [reply core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java]: I think we can throw an exception if possible.

[the remark this answers is not in the document]
Comment (zhengyangyong) [reply core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java]: so, we can process this issue two way :
1. throw a ServiceCombException exception, not NPE
2. Give a error log to let user know it is not an appropriate practice and fix as quick as possible, and let service go

This issue come from user report, he want fill interface later

[the remark this answers is not in the document]
Comment (liubao68) [reply core/src/main/java/org/apache/servicecomb/core/definition/SchemaMeta.java]: OK. In this way, we need to change the message to warning.

**Pull request merged**